### PR TITLE
Php02db

### PR DESCRIPTION
**Pull request by rakeshmindfire**

### CODE REVIEW

**Comment by rakeshmindfire** [comment]
demo request

File: css/style.css [hunk 1]
```diff
@@ -4,16 +4,27 @@
...
 }
 
-.navbar-brand{
+.navbar-brand {
     font-size: 28px;
-}
\ No newline at end of file
+}
+
+give_padding {
```
Comment (naincy) [inline css/style.css]: use consistent naming convention either give_padding or navbar-brand

File: css/style.css [hunk 2]
```diff
@@ -4,16 +4,27 @@
...
+    padding-top: 2%;
+}
+
+table img {
+    width: 100px;
+    height: 100px;
+}
+td, th {
```
Comment (naincy) [inline css/style.css:29]: here should be line break.

File: helper/validation.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
+<?php
+
+function image_check($pic){
+    
+    $uploadOk = 1;
```
Comment (naincy) [inline helper/validation.php]: Its Good habit to use Boolean(TRUE/FALSE) values in place of using 0 or 1

File: helper/validation.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    
+    if($_FILES[$pic]['error']==0){ // $FILES['name']['error'] is 0 if successfully uploaded
```
Comment (naincy) [inline helper/validation.php]: - Remove dead code
- Use === operator in place of == (we discussed on this)
- format your code
  if($_FILES[$pic]['error']==0){ to if ($_FILES[$pic]['error'] === 0) {

File: helper/validation.php [hunk 5]
```diff
@@ -0,0 +1,47 @@
+<?php
+
+function image_check($pic){
```
Comment (naincy) [inline helper/validation.php]: Add function header and also rename the function with proper name like validate_image or image_validation

File: helper/validation.php [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        //    echo "Sorry, your file is too large.";
+            $uploadOk = 0;
+        }
+
+        if(!in_array($extension, array('jpeg','jpg','png','gif'))) {
+        //   echo "Sorry, only JPG, JPEG, PNG & GIF files are allowed.";
+           $uploadOk = 0;
+        }
+    } 
+return $uploadOk;
```
Comment (naincy) [inline helper/validation.php]: Return proper error msg if image not valid in place of returning only upload is ok or not

File: helper/validation.php (hunk 3 above)
Comment (naincy) [inline helper/validation.php]: Keep variable names consistent
variable $uploadOk should be like $upload_ok or $is_upload_allowed or $is_image_valid

File: helper/validation.php [hunk 7]
```diff
@@ -0,0 +1,47 @@
+<?php
+
+function image_check($pic){
+    
+    $uploadOk = 1;
+    
+    if($_FILES[$pic]['error']==0){ // $FILES['name']['error'] is 0 if successfully uploaded
+    
+        $extension = pathinfo( basename($_FILES[$pic]["name"]))['extension'];
```
Comment (naincy) [inline helper/validation.php]: Use single quote(') and make consistency in project
 $extension = pathinfo(basename($_FILES[$pic]['name']))['extension'];

DO changes in all places where ever is required

File: index.php [hunk 8]
```diff
@@ -0,0 +1,39 @@
+<!DOCTYPE html>
+<html lang="en">
+
```
Comment (naincy) [inline index.php:3]: Remove unnecessary line break

File: index.php [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+<head>
+
+    <title>QuickSeller : Home</title>
+
+    <?php
+        require_once 'templates/header.php';     
+     ?>
+
```
Comment (naincy) [inline index.php]: Remove unnecessary line break

File: index.php [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+        <div class="container">
+            <div class="intro-text">
+                <div class="intro-lead-in">Welcome To QuickSeller</div>
+                <div class="intro-heading">1,2,3 and its sold</div>
+                <a href="log_in.php" class=" btn btn-xl">Log in</a>
+            </div>
+        </div>
+    </header>
+
+
```
Comment (naincy) [inline index.php:61]: Remove unnecessary line break

File: index.php [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+    <!-- jQuery -->
+    <script src="vendor/jquery/jquery.min.js"></script>
+
+    <!-- Bootstrap Core JavaScript -->
+    <script src="vendor/bootstrap/js/bootstrap.min.js"></script>
+
+</body>
+
```
Comment (naincy) [inline index.php]: Remove unnecessary line break

File: product_list.php [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+$conn = mysqli_connect(SERVERNAME, USERNAME, PASSWORD, DBNAME);
+
+// Handled case if connection failed
+if (!$conn) {
+    die("Connection failed: " . mysqli_connect_error());
+}
+
+if(isset($_GET['delete_id']) ) {
+    $sql_delete="DELETE FROM `products_list` WHERE `products_list`.`id` ='". $_GET['delete_id']."'";
+    mysqli_query($conn, $sql_delete);
```
Comment (naincy) [inline product_list.php]: Case not handled
- if execution of query fails

File: product_list.php [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+if (!$conn) {
+    die("Connection failed: " . mysqli_connect_error());
+}
+
+if(isset($_GET['delete_id']) ) {
+    $sql_delete="DELETE FROM `products_list` WHERE `products_list`.`id` ='". $_GET['delete_id']."'";
+    mysqli_query($conn, $sql_delete);    
+} 
+
+
```
Comment (naincy) [inline product_list.php]: Remove unnecessary line break

File: product_list.php [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+
+$sql_getproducts="SELECT pl.id,pc.name as category_name,pl.image,pl.name as product_name,pl.amount,"
```
Comment (naincy) [inline product_list.php]: - Variable name should be $sql_get_products
- SELCTE SQL should be in proper format like
  $sql_get_products = 'SELECT pl.id,pc.name as category_name, pl.image, pl.name as product_name, pl.amount, '
  . 'pl.description, pl.created_date'
  . 'FROM products_list pl '
  . 'JOIN products_category pc ON pl.category=pc.id'
  . 'WHERE pl.user_id=33 '
  . 'ORDER BY pl.created_date DESC';

File: product_list.php [hunk 15]
```diff
@@ -0,0 +1,93 @@
...
+        . "pl.description,pl.created_date FROM products_list pl "
+        . "JOIN products_category pc ON pl.category=pc.id WHERE pl.user_id=33 "
+        . "ORDER BY pl.created_date DESC" ;
+echo $sql_getproducts;
```
Comment (naincy) [inline product_list.php]: Why echo of $sql_getproduct is required here?

File: product_list.php [hunk 16]
```diff
@@ -0,0 +1,93 @@
...
+
+$products = mysqli_query($conn, $sql_getproducts);
+
+
```
Comment (naincy) [inline product_list.php:32]: - Case missing if execution of query fails
- Remove unnecessary line break

File: product_list.php [hunk 17]
```diff
@@ -0,0 +1,93 @@
...
+            <th>Name</th>
+            <th>Amount</th>
+            <th>Description</th>
+            <th>Uploaded on</th>
+            <th>Actions</th>
+          </tr>
+        </thead>
+        <tbody>
+            <?php
+                if(mysqli_num_rows($products)>0){
```
Comment (naincy) [inline product_list.php]: Handling only tbody contents if no products present.
Please handle complete table, if no products present do not show table.
Can show some message like 'No products found to list.'

File: product_register.php [hunk 18]
```diff
@@ -0,0 +1,210 @@
...
+if (!$conn) {
+    die("Connection failed: " . mysqli_connect_error());
+}
+
+if(isset($_GET['update_id']) ) {
+    $is_update = TRUE;
+    $sql_getvaluestoupdate="SELECT `category`,`name`,`amount`,`description` FROM `products_list` "
+        . "WHERE `products_list`.`id` ='". $_GET['update_id']."'";
+    $row_toupdate=mysqli_fetch_assoc(mysqli_query($conn, $sql_getvaluestoupdate));
+    print_r ($row_toupdate);
```
Comment (naincy) [inline product_register.php]: Is print_r required here?

File: product_register.php [hunk 19]
```diff
@@ -0,0 +1,210 @@
...
+$conn = mysqli_connect(SERVERNAME, USERNAME, PASSWORD, DBNAME);
+
+// Handled case if connection failed
+if (!$conn) {
+    die("Connection failed: " . mysqli_connect_error());
+}
+
+if(isset($_GET['update_id']) ) {
+    $is_update = TRUE;
+    $sql_getvaluestoupdate="SELECT `category`,`name`,`amount`,`description` FROM `products_list` "
```
Comment (naincy) [inline product_register.php]: Variable name $sql_getvaluestoupdate is not consistent

**Comment by naincy** [comment]
Basic code guidelines are missing
- Variable name inconsistency
- single quote ('') and double quote("")  inconsistency
- Line breaks
- comments are missing in logical does
- Remove Dead code where its is present
- Remove Debugging code
- Cases missing for execution failure
- Function Header is missing

Need to review again once you fix with these basic guidelines

**Pull request closed without merging**